pennylane/devices/qubit/apply_operation.py

```diff
@@ -407,14 +407,83 @@ def apply_pauliz(op: qml.Z, state, is_state_batched: bool = False, debugger=None
     return math.stack([state[sl_0], state1], axis=axis)
 
 
+@apply_operation.register
+def apply_phaseshift(op: qml.PhaseShift, state, is_state_batched: bool = False, debugger=None, **_):
+    """Apply PhaseShift to state."""
+
+    n_dim = math.ndim(state)
+
+    if n_dim >= 9 and math.get_interface(state) == "tensorflow":  # pragma: no cover
```

Comment:
Will be a bit more expensive and slower, but it's probably worth a test to make sure that these operations don't fall over on 10 qubit state vectors.

Reply:
Will do.